### ISSUE TITLE
Can't show complete info in flame graph.

### ISSUE THREAD

**Issue by futurehome** [KDAB/hotspot#630]
**Describe the bug**
The first time I use hotspot, I got the flamegraph as expected as below.
![1714447562596_5DA6B62F-446D-4a30-B354-655E02600E68](https://github.com/KDAB/hotspot/assets/6230304/a1fe7031-445e-4e89-b2dc-6b36ce902968)

Lator on, with the same code, and same run time, same setting,  I can only get flamegraph like below, there are a lot of funcs missing, anything wrong?
![image](https://github.com/KDAB/hotspot/assets/6230304/76425dd2-d50a-46a3-a04b-2c75c2613aaa)

**To Reproduce**

**Expected behavior**

**Screenshots**

**Version Info (please complete the following information):**

- Linux Kernel version:
- perf version:
- hotspot version (appimage? selfcompiled?):
- if self-compiled hotspot, what version of elfutils:

**Additional context**

**Comment by lievenhey** [comment]
Which version are you using?

**Comment by milianw** [comment]
Export the data as a .perfparser file and give explicit steps needed to reproduce this

**Comment by HybridDog** [comment]
I may have the same problem. I use hotspot-v1.5.1-x86_64.AppImage. 
Hotspot shows a lot of error messages and there are more `Invalid memory read requested by dwfl fffffffffffffff8` messages when the flamegraph misses information. I don't know if the error messages are related.
[broken_stderr.txt](https://github.com/user-attachments/files/15959656/broken_stderr.txt)
[working_stderr.txt](https://github.com/user-attachments/files/15959660/working_stderr.txt)
When I opened the file generated by perf the first time, the flame graph looked fine, but when  opened it again, parts were missing. When I opened it again with hotspot-v1.4.1-301-g271a781-x86_64.AppImage or hotspot-continuous.AppImage, the parts were still missing.
